pkg/providers/azure_test.go

```diff
@@ -139,109 +145,124 @@ func TestGetTenantId_NilTenantInList(t *testing.T) {
 	}
 }
 
-func TestGetAppObjectId(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	mockGraphClient := mock_providers.NewMockGraphClient(ctrl)
-
-	appID := "testAppID"
-	expectedAppID := "mockAppID"
-	mockGraphClient.EXPECT().GetApplicationObjectId(gomock.Any(), appID).Return(expectedAppID, nil)
+var testAppID = "mockAppID"
+var testID = "mockID"
+var errToSend error = nil
 
-	sc := &SetUpCmd{
-		appId: appID,
-		AzClient: AzClient{
-			GraphClient: mockGraphClient,
-		},
-	}
+type mockApplicationable struct {
+	models.Applicationable
+}
 
-	err := sc.getAppObjectId(context.Background())
+func (m *mockApplicationable) GetAppId() *string {
+	return &testAppID
+}
 
-	if err != nil {
-		t.Errorf("Expected no error, got: %v", err)
-	}
-	if sc.appObjectId != expectedAppID {
-		t.Errorf("Expected application ID %s, got: %s", expectedAppID, sc.appObjectId)
-	}
+func (m *mockApplicationable) GetId() *string {
+	return &testID
 }
 
-func TestGetAppObjectId_Error(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
+type mockSerialWriter struct {
```

Comment:
why is `mockSerialWriter` struct needed here?

Reply:
The Post() function in createAzApp() uses 'requestInfo' which in turn calls the 'ToPostRequestInformation' and that calls the 'SetContentFromParsable' and which calls the  getSerializationWriter as shown below. Since we don't have an interface for the Application() I had to mock a bunch of functions.
writer, err := request.getSerializationWriter(requestAdapter, contentType, item)
	if err != nil {
		span.RecordError(err)
		return err
	}

Reply:
okay makes sense